src/app/ui/editor/moving_pixels_state.cpp

```diff
@@ -532,6 +539,20 @@ void MovingPixelsState::onBeforeCommandExecution(CommandExecutionEvent& ev)
   if (!isActiveEditor())
     return;
 
+  // Abort if layer or layer hierarchy is locked
+  Layer* layer = m_editor->getSite().layer();
+  if (layer && !layer->isEditableHierarchy() &&
+      (command->id() == CommandId::MoveMask() ||
+      command->id() == CommandId::Flip() ||
+      command->id() == CommandId::Cut() ||
+      command->id() == CommandId::Clear() ||
+      command->id() == CommandId::Rotate())) {
+    StatusBar::instance()->showTip(
+      1000, fmt::format("Layer '{}' is locked", layer->name()));
+    dropPixels();
```

Comment:
I'm thinking if we should `dropPixels()` here or listening for changes in the lock status of the layer i.e. when the lock changes, we drop the pixels, so we are not going to be here in MovingPixelsState with an unlocked layer. Maybe when a "on before command" of `LayerLock` is received, we drop the pixels if the layer is unlocked.